src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -517,6 +517,19 @@ public T WithTelemetry(ITelemetryConfig telemetryConfig)
             return (T)this;
         }
 
+        /// <summary>
+        /// Configures MSAL to retry on 5xx server errors. When enabled (on by default), MSAL will wait 1 second after recieving a 5xx error and then retry the
+        /// http request again.
+        /// </summary>
+        /// <param name="shouldRetry">Determines if MSAL should retry on 5xx errors.</param>
+        /// <returns></returns>
+        public T WithRetryOnServerErrors(bool shouldRetry)
```

Comment:
in the end did we decide not to add it to:

```csharp
WithHttpClientFactory(IHttpClientFactory factory, RetryOptions retryOptions).
```

@bgavrilMS: I thought that's what we agreed on? Do I miss something?

Reply:
We decided to not provide a complex retry option and to focus only on: 

```csharp
WithHttpClientFactory(IHttpClientFactory factory, bool retryOnceOn5xx=true).
```

Or did you want to have a `RetryOptions` object with a single `bool RetryOnceOn5xx` property, as it would be easier to extend?